docs/api-contract/start/contract.read.md

```diff
@@ -12,28 +12,33 @@ import { ContractPromise } from '@polkadot/api-contract';
 // string or the raw JSON data (after doing a JSON.parse). The address is
 // the actual on-chain address as ss58 or AccountId object.
 const contract = new ContractPromise(api, abi, address);
-
-// Read from the contract
-...
 ```
 
-Either via a create above or via a call to `createContract` both instances are the same. The `Contract` provides a wrapper around the `Abi` and allows you to call either `read` or `exec` on a contract to interact with it.
-
-
 ## Reading contract values
 
-In the `Blueprint` example we have instantiated an incrementer contract. In the following examples we will continue using it to read from and execute transactions into, since it is a well-known entity. To read a value from the contract, we can do the following -
+Under the hood `.query.<messageName>` is using the `api.rpc.contracts.call` API on the contracts pallet to retrieve a result. For this interface, the format is always of the form `messageName(<account address to use>, <value>, <gasLimit>, <...additional params>)`.
 
 ```javascript
-// Read from the contract via an RPC call
-const value = 0; // only useful on isPayable messages
 
-// NOTE the apps UI specified these in mega units
-const gasLimit = 3000n * 1000000n;
+// Contract query options
+const options = { 
+  // maximum gas to be consumed for the contract execution dry run. if limit is too small the dry run will fail. 
+  // you can use -1 use the maximum gas limit available to transactions
+  gasLimit,
```

Comment:
Might be better to keep the sample values here? Also to use them in the other cases where they aren't defined at all.